context/app/static/js/components/Detail/DetailLayout.jsx

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable camelcase */
+import React from 'react';
+import styled from 'styled-components';
+import NoticeAlert from '../NoticeAlert';
+import TableOfContents from './TableOfContents';
+
+const FlexColumn = styled.div`
+  flex-grow: 1;
+  display: flex;
+  flex-direction: column;
+  max-width: 1280px;
+`;
+
+const FlexRow = styled.div`
+  flex-grow: 1;
+  display: flex;
+  justify-content: center;
+`;
+
+function DetailLayout(props) {
+  const {
+    displayViz,
+    displayProtocol,
+    displayMetadata,
+    displayMetadataTable,
+    displayFiles,
+    displayDag,
+    flashed_messages,
+    children,
+  } = props;
+
+  const updateSections = () => {
+    const sections = new Map([
+      ['summary', { text: 'Summary', hash: 'summary' }],
```

Comment:
Make a little function for this? If you really want dagProv to be a special case, ok, but in all the others, it would be hard to spot typos.